sparkles/roll_optimize.py

```diff
@@ -367,8 +366,19 @@ def get_roll_options(
 
         att_targ = self.att_targ
 
-        # Special case, first roll option is self but with obsid set to roll
-        acar = deepcopy(self)
+        # Special case, first roll option is self but with obsid set to roll. Note that
+        # the "copy" from the line below is not a full deep copy of the table meta
+        # because it runs the __init__ method. This initializes the MetaAttributes to
+        # their default states instead of copying them. This is *required* for this code
+        # to work.
+        #
+        # As a development note, this used to be deepcopy(self), but in astropy 7.0 the
+        # implementation changed so that the meta dict was explicitly deepcopied after
+        # the __init__ call. This caused the MetaAttributes to be set to the original
+        # self values and the code to fail. The line below is the astropy 6.0 code for
+        # implementing deepcopy().
+        acar = self.copy(copy_data=True)
```

Comment:
is this behavior going to be fixed upstream? I would expect that meta is copied if I were to call `copy.deepcopy`.

Of course, this does not matter for this PR. Just asking.

Reply:
Meta is indeed deepcopied for a `deepcopy()` call, so astropy is doing the right thing. The copying of `meta` is actually the *problem* here, because the sparkles roll optimize code expects the `MetaData` attributes to be initialized to the default values, not to inherit from the original.

If you are still confused, this is why it took me several hours to really get to the bottom of this. The root cause is that the `ACAReviewTable` class is a bit unorthodox in that it sets the `MetaData` properties explicitly in the `__init__`. This introduces an ambiguity about copying: should the copy get the initial value from `__init__` or the copied value? Sparkles is expecting the initial value.

Overall, I agree this is a bit of a mess.

Reply:
I wonder if there is a way to make it so the values are set in `Table.__new__`, something like dding an argument that is only used by `Table.__new__`, that is `False` by default, but `Table.__deepcopy__` sets it to `True` when constructing the copy. I think that would run before `uper().__init_()`

Reply:
@javierggt - I'm not quite following you. What would this new argument be controlling? It also sounds like you are implying an upstream change to `Table.__deepcopy__` for this, but that doesn't quite make sense. Of course the subclass in sparkles can override `__deepcopy__` if needed, but actually `deepcopy()` is currently behaving like I would expect (making an exact full copy).

Reply:
Maybe a better thing would be to punt on making the copy and instead just reselect the catalog using the original `call_args` and hope that works.

Reply:
Final discussion, consider changing above to a deepcopy and then do not re-run check_catalog().